src/main/java/it/gov/pagopa/payment/controller/TransactionControllerImpl.java

```diff
@@ -0,0 +1,21 @@
+package it.gov.pagopa.payment.controller;
+
+import it.gov.pagopa.common.performancelogger.PerformanceLog;
+import it.gov.pagopa.payment.dto.TransactionDTO;
+import it.gov.pagopa.payment.service.TransactionService;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class TransactionControllerImpl implements TransactionController {
+    private final TransactionService transactionService;
+
+    public TransactionControllerImpl(TransactionService transactionService) {
+        this.transactionService = transactionService;
+    }
+
+    @Override
+    @PerformanceLog("GET_TRANSACTION")
+    public TransactionDTO getTransaction(String transactionId, String userId) {
```

Comment:
you could return directly TransactionInProgress, there is not need to return a separate DTO :)
Actually this API is an internal API, so the model is not forced to have an exact shape